src/connector/src/sink/file_sink/mod.rs

```diff
@@ -0,0 +1,230 @@
+        // Todo: after enabling sink decouple, use the new naming convention.
+        let epoch = self.epoch.ok_or_else(|| {
+            SinkError::Opendal("epoch has not been initialize, call `begin_epoch`".to_string())
+        })?;
+        if self.sink_writer.is_none() {
+            self.create_sink_writer(epoch).await?;
+        }
+        if self.is_append_only {
+            self.append_only(chunk).await
+        } else {
+            // currently file sink only supports append only mode.
+            unimplemented!()
+        }
+    }
+
+    async fn begin_epoch(&mut self, epoch: u64) -> Result<()> {
+        self.epoch = Some(epoch);
+        Ok(())
+    }
+
+    async fn abort(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    /// For the file sink, currently, the sink decoupling feature is not enabled.
+    /// When a checkpoint arrives, the force commit is performed to write the data to the file.
+    /// In the future if flush and checkpoint is decoupled, we should enable sink decouple accordingly.
+    async fn barrier(&mut self, is_checkpoint: bool) -> Result<()> {
+        if is_checkpoint && let Some(sink_writer) = self.sink_writer.take() {
+            match sink_writer {
+                FileWriterEnum::ParquetFileWriter(w) => {
+                    let _ = w.close().await?;
+                }
+                FileWriterEnum::FileWriter(mut w) => w.close().await?,
+            };
+        }
+
+        Ok(())
+    }
+
+    async fn update_vnode_bitmap(&mut self, _vnode_bitmap: Arc<Bitmap>) -> Result<()> {
+        Ok(())
+    }
+}
+
+impl OpenDalSinkWriter {
+    pub fn new(
+        operator: Operator,
+        write_path: &str,
+        rw_schema: Schema,
+        pk_indices: Vec<usize>,
+        is_append_only: bool,
+        executor_id: u64,
+        encode_type: SinkEncode,
+    ) -> Result<Self> {
+        let arrow_schema = convert_rw_schema_to_arrow_schema(rw_schema)?;
+        Ok(Self {
+            schema: Arc::new(arrow_schema),
+            write_path: write_path.to_string(),
+            pk_indices,
+            operator,
+            sink_writer: None,
+            is_append_only,
+            epoch: None,
+            executor_id,
+            encode_type,
+        })
+    }
+
+    async fn create_object_writer(&mut self, epoch: u64) -> Result<OpendalWriter> {
+        // Todo: specify more file suffixes based on encode_type.
+        let suffix = match self.encode_type {
+            SinkEncode::Json => "json",
+            SinkEncode::Parquet => "parquet",
+            _ => unimplemented!(),
+        };
+
+        // Note: sink decoupling is not currently supported, which means that output files will not be batched across checkpoints.
+        // The current implementation writes files every time a checkpoint arrives, so the naming convention is `epoch + executor_id + .suffix`.
+        let object_name = format!(
+            "{}/epoch_{}_executor_{}.{}",
```

Comment:
I suggest we don't include `epoch` and `executor` keywords in the file name because user doesn't need and doesn't want to understand that. We can just generate an opaque id in the name here: `format!("{}/{}_{}.{}", self.write_path, epoch, self.executor_id, suffix)`